PHP parse scripts/Modified_php_parse_scripts_by_Baertram/game_api.php

```diff
@@ -90,34 +95,72 @@ public function parseClasses($array)
                     $parts = explode(",", $matches['params']);
                     foreach ($parts as $part) {
                         $matches2 = null;
-                        $partClean = str_replace('function', 'functionName', $part);
 
                         //* CallSecureProtected(*string* _functionName_, *types* _arguments_)
                         //* IsTrustedFunction(*function* _function_)
-                        if (preg_match('/\*(?P<type>.*)?\* _(?P<param>.*?)_/', $partClean, $matches2)) {
-                            $objects[$methodClean]['params'][$matches2['param']] = $matches2['type'];
+                        if (preg_match('/\*(?P<type>.*)?\* _(?P<param>.*?)_/', $part, $matches2)) {
+                            $type = $this->processType($matches2['type']);
+                            if ($type == '...') {
+                                $matches2['param'] = '...';
+                                $type = 'any';
+                            }
+                            $objects[$methodClean]['params'][$matches2['param']] = $type;
                         }
                     }
                 }
 
+                if (strpos($line, '_Uses variable returns..._') !== false) {
+                    $variableReturns = true;
+                }
+
                 $matches = null;
                 if (preg_match('/\*\* _Returns\:_ (?P<parts>.*)/', $line, $matches)) {
                     $parts = explode(",", $matches['parts']);
                     foreach ($parts as $part) {
                         $matches2 = null;
-                        $partClean = str_replace('function', 'functionName', $part);

```

Comment:
Also I'd propose to use |nil instead of the ? for params that could be nilable.
The ? is not known if you do not code with vs (afaik) and having the nil there should be clear.
According to others number|nil should be accepted by vsCode and LuaLS too

Reply:
> Also I'd propose to use |nil instead of the ? for params that could be nilable. The ? is not known if you do not code with vs (afaik) and having the nil there should be clear. According to others number|nil should be accepted by vsCode and LuaLS too

LuaLS should understand ? in any IDE, but fair, I'll change it to |nil